config/openapi-generator.json

```diff
@@ -5,5 +5,6 @@
     "inlineSchemaOptions": {
         "RESOLVE_INLINE_ENUMS": true
     },
-    "enumClassPrefix": true
+    "enumClassPrefix": true,
+    "useOneOfDiscriminatorLookup": true
```

Comment:
Actually I'm misunderstanding the edge case.  The response _will_ be validated against the matched schema: https://github.com/OpenAPITools/openapi-generator/blob/638af0fe37e2786ca3d6f5a8d764b346b2cc74a6/modules/openapi-generator/src/main/resources/go/model_oneof.mustache#L48

What _won't_ be validated is that the response doesn't match multiple members of the `oneOf` schema; that makes me even less concerned since arguably the point of the discriminator in OpenAPI 3 is to be the thing that matches one member schema.

Reply:
CI wins this round: https://github.com/equinix/equinix-sdk-go/actions/runs/9277979581/job/25528195329?pr=59#step:5:6736

This config change impacts the generated code for other services, and CI is failing because those code changes haven't been committed as part of this PR.  I tried it out locally and see that this config change makes a material difference for other services.  In the case of `metalv1`, at least, it results in losing strict unmarshalling that we may not want to lose.

For now it's probably better to set the `useOneOfDiscriminatorLookup` property to true in the `codegen` task in `Makefile.eiav2` to limit the scope, and then open a GitHub issue proposing to adopt that more broadly so we can revisit.  It's possible the code change I'm seeing in `metalv1` is a bug in the generator.

Reply:
Totally fair point. Glad that identified it. I've moved it from global to local and repushed. Ran the makefile again and it generates exactly the same as if the config was in place globally. 

It would make changes to any other SDK that was generated with a discriminator in the spec. I had overlooked that. A little too narrow in the focus but I'll have to keep that in mind because we're housing multiple projects here.